AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_CubeMapCaptureAdded.py

```diff
@@ -0,0 +1,116 @@
+        "CubeMap Capture Entity successfully created",
+        "P0: CubeMap Capture Entity failed to be created")
+    cubemap_capture_component = (
+        "Entity has a CubeMap Capture component",
+        "P0: Entity failed to find CubeMap Capture component")
+    cubemap_capture_component_removal = (
+        "UNDO CubeMap Capture Entity success",
+        "UNDO CubeMap Capture Entity failed")
+    removal_undo = (
+        "REDO CubeMap Capture Entity success",
+        "REDO CubeMap Capture Entity failed")
+    entity_deleted = (
+        "Entity deleted",
+        "P0: Entity was not deleted")
+    deletion_undo = (
+        "UNDO Entity deletion success",
+        "P0: UNDO Entity deletion failed")
+    deletion_redo = (
+        "REDO Entity deletion success",
+        "P0: REDO Entity deletion failed")
+
+
+def AtomEditorComponents_CubeMapCapture_AddedToEntity():
+    """
+    Summary:
+    Tests the Cubemap Capture component can be added to an entity and has the expected functionality.
+
+    Test setup:
+    - Wait for Editor idle loop.
+    - Open the "Base" level.
+
+    Expected Behavior:
+    The component can be added, and deleted.
+    Creation and deletion undo/redo should also work.
+
+    Test Steps:
+    1) Create a Cubemap_Capture entity with no components.
+    2) Add Cubemap_Capture component to Cubemap_Capture entity.
+    3) Remove the Cubemap_Capture component.
+    4) UNDO the Cubemap_Capture component removal.
+    5) Delete the Cubemap_Capture entity.
+    6) UNDO deletion.
+    7) REDO deletion.
+    8) Look for errors.
+
+    :return: None
+    """
+
+    import azlmbr.legacy.general as general
+
+    from editor_python_test_tools.editor_entity_utils import EditorEntity
+    import editor_python_test_tools.prefab_utils as PrefabUtils
+    from editor_python_test_tools.utils import Report, Tracer, TestHelper
+    from Atom.atom_utils.atom_constants import AtomComponentProperties
+
+    with Tracer() as error_tracer:
+        # Test setup begins.
+        # Setup: Wait for Editor idle loop before executing Python hydra scripts then open "Base" level.
+        TestHelper.init_idle()
+        TestHelper.open_level("Graphics", "base_empty")
+
+        # Test steps begin.
+        # 1. Create a Cubemap_Capture entity with no components.
+        cubemap_capture_entity = EditorEntity.create_editor_entity(AtomComponentProperties.cube_map_capture())
+        Report.critical_result(Tests.cubemap_capture_creation, cubemap_capture_entity.exists())
+
+        # 2. Add Cubemap_Capture component to Cubemap_Capture entity.
+        cubemap_capture_component = cubemap_capture_entity.add_component(AtomComponentProperties.cube_map_capture())
+        Report.critical_result(Tests.cubemap_capture_component, cubemap_capture_entity.has_component(AtomComponentProperties.cube_map_capture()))
+
+        # 3. Remove the Cubemap_Capture component.
+        cubemap_capture_component.remove()
+        PrefabUtils.wait_for_propagation()
```

Comment:
just curious why wait_for_propagation() when this isn't a prefab?

Reply:
I was using the hydra_AtomEditorComponents_BloomAdded.py for a reference. For the Undo deletion and Redo deletion it used: 
```

        # 44. UNDO deletion.
        general.undo()
        PrefabUtils.wait_for_propagation()
        Report.result(Tests.deletion_undo, bloom_entity.exists())

        # 45. REDO deletion.
        general.redo()
        PrefabUtils.wait_for_propagation()
        Report.result(Tests.deletion_redo, not bloom_entity.exists())
        
```

The other version undo redo version I saw was this 
```

# 3. Remove the Bloom component.
        bloom_component.remove()
        general.idle_wait_frames(1)
        Report.critical_result(Tests.bloom_component_removal,
                               not bloom_entity.has_component(AtomComponentProperties.bloom()))

        # 4. Undo Bloom component removal.
        general.undo()
        general.idle_wait_frames(1)
        Report.result(Tests.removal_undo, bloom_entity.has_component(AtomComponentProperties.bloom()))

```

Would the second version with general.idle_wait_frames be better?

Reply:
I'm not sure prefab team consulted us directly when changing our tests. we aren't doing prefabs here so I prefer idle_wait_frames since their wait_for_propagation is actually just a wrapper on idle_wait_frames(1)

Reply:
Just adding idle wait frames without adding a comment as to why the wait is needed isn't good in my opinion. It makes understanding and refactoring the tests harder. With wait_for_propagation(), the name at least explains why the wait is needed. 

@smurly , I can try to shed some light on why wait_for_propagation() is needed here : When prefabs are enabled (which is the default for the editor), undo and redo operations are purely dom authoritative. It means that they update the underlying Dom values but those changes don't make it to actual entities. An additional frame wait is needed for Dom values to be propagated to the entities in the level. I'd go as far and say that there can be a common undo/redo helper function across all tests that always waits for propagation when undo and redo are called.